clients/src/main/java/org/apache/kafka/common/security/plain/internals/PlainSaslServer.java

```diff
@@ -81,14 +82,11 @@ public PlainSaslServer(CallbackHandler callbackHandler) {
          *                ;; any UTF-8 encoded Unicode character except NUL
          */
 
-        String[] tokens;
-        try {
-            tokens = new String(response, "UTF-8").split("\u0000");
-        } catch (UnsupportedEncodingException e) {
-            throw new SaslException("UTF-8 encoding not supported", e);
-        }
-        if (tokens.length != 3)
+        String response = new String(responseBytes, StandardCharsets.UTF_8);
+        String[] tokens = split(response, "\u0000");
+        if (tokens.length != 3) {
             throw new SaslException("Invalid SASL/PLAIN response: expected 3 tokens, got " + tokens.length);
```

Comment:
Do we still need this?

Reply:
Good point. We can add code to `split` (and maybe rename it) to verify if two utf8nul are not present and return an error like this one. Would it be `SaslException` still?

Reply:
I think this can also be `SaslAuthenticationException` since it is only returning error based on the information the client passed in, so it wont be leaking any server state.

Reply:
Sounds good.